### ISSUE TITLE
[Question] Closing the loops on model export

### ISSUE THREAD

**Issue by chunky** [DLR-RM/stable-baselines3#673]
## Question

I'm trying to export a model to TFLite, and so far, following the guidance in the documentation, I'm falling short, there are errors during the actual export

Here's where I'm at:

### Create a model:
1. Using a Gym closely related in obs/action to the custom one I'm working with
2. This won't learn anything useful, but it does create the model file
3. I've had most success with my gym using SAC. When using PPO, I have other problems following the export.

```python
import gym
import numpy as np

from stable_baselines3 import SAC

env_name = "MountainCarContinuous-v0"
model_save_file = "model.zip"
env = gym.make(env_name)

model = SAC('MlpPolicy', env, verbose=1)
model.learn(total_timesteps=30)
model.save(model_save_file)
```

### Convert model
1. This is lifted primarily, and in spirit, from the documentation
2. The error I get is in the bowels of the convert, in what looks like an attempt to trace the forward pass [entire stack trace pasted below, final snip here]:
```
in "torch/nn/functional.py", line 1849, in linear
TypeError: linear(): argument 'input' (position 1) must be Tensor, not list"
```

```python
import gym
import torch

from stable_baselines3 import SAC

class OnnxablePolicy(torch.nn.Module):
  def __init__(self,  actor):
      super(OnnxablePolicy, self).__init__()
      self.actor = torch.nn.Sequential(actor.latent_pi, actor.mu)

  def forward(self, observation):
      # NOTE: You may have to process (normalize) observation in the correct
      #       way before using this. See `common.preprocessing.preprocess_obs`
      return self.actor(observation)

env_name = "MountainCarContinuous-v0"
model_save_file = "model.zip"
onnx_save_file = "model.onnx"
env = gym.make(env_name)

model = SAC.load(model_save_file, env, verbose=True)
obs = env.observation_space
# tolist() is because it's a numpy array
dummy_input = obs.sample().tolist()

onnxable_model = OnnxablePolicy(model.policy.actor)
model.policy.to("cpu")
model.policy.eval()

torch.onnx.export(onnxable_model, dummy_input, onnx_save_file, opset_version=9, verbose=True)
```

## Additional context

Here's the full stack trace where it goes sideways:
```
(venv) chunky@chunklap:/mnt/sup2rtam/sb3_to_coral$ python3 minimal_conv.py 
Wrapping the env with a `Monitor` wrapper
Wrapping the env in a DummyVecEnv.
Traceback (most recent call last):
  File "/mnt/sup2rtam/sb3_to_coral/minimal_conv.py", line 30, in <module>
    torch.onnx.export(onnxable_model, dummy_input, onnx_save_file, opset_version=9, verbose=True)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/onnx/__init__.py", line 316, in export
    return utils.export(model, args, f, export_params, verbose, training,
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/onnx/utils.py", line 107, in export
    _export(model, args, f, export_params, verbose, training, input_names, output_names,
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/onnx/utils.py", line 724, in _export
    _model_to_graph(model, args, verbose, input_names,
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/onnx/utils.py", line 493, in _model_to_graph
    graph, params, torch_out, module = _create_jit_graph(model, args)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/onnx/utils.py", line 437, in _create_jit_graph
    graph, torch_out = _trace_and_get_graph_from_model(model, args)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/onnx/utils.py", line 388, in _trace_and_get_graph_from_model
    torch.jit._get_trace_graph(model, args, strict=False, _force_outplace=False, _return_inputs_states=True)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/jit/_trace.py", line 1166, in _get_trace_graph
    outs = ONNXTracedModule(f, strict, _force_outplace, return_inputs, _return_inputs_states)(*args, **kwargs)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/jit/_trace.py", line 127, in forward
    graph, out = torch._C._create_graph_by_tracing(
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/jit/_trace.py", line 118, in wrapper
    outs.append(self.inner(*trace_inputs))
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1090, in _slow_forward
    result = self.forward(*input, **kwargs)
  File "/mnt/sup2rtam/sb3_to_coral/minimal_conv.py", line 15, in forward
    return self.actor(observation)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1090, in _slow_forward
    result = self.forward(*input, **kwargs)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/nn/modules/container.py", line 141, in forward
    input = module(input)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1090, in _slow_forward
    result = self.forward(*input, **kwargs)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/nn/modules/container.py", line 141, in forward
    input = module(input)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1102, in _call_impl
    return forward_call(*input, **kwargs)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/nn/modules/module.py", line 1090, in _slow_forward
    result = self.forward(*input, **kwargs)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/nn/modules/linear.py", line 103, in forward
    return F.linear(input, self.weight, self.bias)
  File "/mnt/sup2rtam/sb3_to_coral/venv/lib/python3.9/site-packages/torch/nn/functional.py", line 1848, in linear
    return torch._C._nn.linear(input, weight, bias)
TypeError: linear(): argument 'input' (position 1) must be Tensor, not list
```

Judicious use of print at that instant, the variable "input" is one of these:
[tensor(-0.2930, dtype=torch.float64), tensor(-0.0676, dtype=torch.float64)]

## Checklist

- [*] I have read the [documentation](https://stable-baselines3.readthedocs.io/en/master/) (**required**)
- [*] I have checked that there is no similar [issue](https://github.com/DLR-RM/stable-baselines3/issues) in the repo (**required**)

**Comment by Miffyli** [comment]
I think the error is here:

```python
dummy_input = obs.sample().tolist()
```

I am not familiar with ONNX, but looking at the doc this should be a torch tensor, but a list.

**Comment by chunky** [comment]
@Miffyli Fabulous!

@araffin It has been a couple decades since I was urged to RTFM. Sincerely, and without sarcasm, I appreciate the reminder.

@Miffyli  , @araffin : I've pulled this together into a few short scripts, hopefully an effective sample for anyone else who wants to tread the path I've walked. The code is all here: https://github.com/chunky/sb3_to_coral . I don't know whether it meets your astonishingly-high standards for documentation [literally, your documentation is stellar], but if so, I proffer it here as something you are welcome to link to in your "exporting models" section.
